### PR TITLE
Find by order_id

### PR DESCRIPTION
**Pull request by ko-eric**
Feature
# find allows you to retrieve a case by orderId

If found it returns a hash of the case.
Otherwise, it will raise Signifyd::InvalidRequestError

### CODE REVIEW

**Comment by grossjo** [comment]
+1

File: lib/signifyd.rb [hunk 1]
```diff
@@ -213,7 +213,9 @@ def self.local_mode
   # @param[Opt]: String[api_key] - 'YOUR-API-KEY'
   # @param[Opt]: Hash[options] - optional parameters
   # @return: Hash - containing response code, body, and other data
-  def self.request(method, url, params, api_key=nil, options={})
+  def self.request(method, url, params={}, api_key=nil, options={})
+    # params  = params.with_indifferent_access
+    # options = options.with_indifferent_access
```
Comment (ssoroka-se) [inline lib/signifyd.rb]: remove or implement

File: lib/signifyd.rb [hunk 2]
```diff
@@ -243,7 +245,9 @@ def self.request(method, url, params, api_key=nil, options={})
     # are necessary, some are not.
     case method.to_s
     when 'get'  || :get
-
+      if options.has_key?(:order_id)
+        url = url.gsub('cases', "orders/#{options[:order_id]}/case")
+      end
```
Comment (ssoroka-se) [inline lib/signifyd.rb:248]: while you're here, get rid of the || :get etc on the other lines. `case method.to_s` will never return a symbol.

File: lib/signifyd/api/list.rb [hunk 3]
```diff
@@ -5,6 +5,11 @@ module ClassMethods
         def all(filters={}, api_key=nil)
           Signifyd.request(:get, self.url, {}, api_key, filters)
         end
+        # To retrieve case by orderId pass in {order_id: value}
+        # into options
+        def find(options={}, filters={},api_key=nil)
+          Signifyd.request(:get, self.url, filters, api_key, options)
```
Comment (ssoroka-se) [inline lib/signifyd/api/list.rb]: self is superfluous.

File: signifyd.gemspec [hunk 4]
```diff
@@ -31,4 +31,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'pry'
   s.add_development_dependency 'simplecov'
   s.add_development_dependency 'coveralls'
+  s.add_development_dependency 'debugger'
```
Comment (ssoroka-se) [inline signifyd.gemspec]: :/ meh

File: spec/lib/signifyd/case_spec.rb [hunk 5]
```diff
@@ -4,14 +4,16 @@
   let(:hash) { SignifydRequests.valid_case }
   let(:json) { JSON.dump(hash) }
   let(:case_id) { (rand * 1000).ceil }
+  let(:order_id) { 'unique_id'}
   let(:investigation) { "{\"investigationId\":#{case_id}}" }
-  
-  context '.create' do    
+  let(:retrieved_case) {"{\"orderId\":#{case_id}}" }
```
Comment (ssoroka-se) [inline spec/lib/signifyd/case_spec.rb:9]: %() can be used in place of "" so that you don't have to escape inner ". just fyi.

**Pull request closed without merging**